cmd/dlv/main.go

```diff
@@ -49,5 +48,92 @@ func main() {
 		os.Exit(0)
 	}
 
-	cli.Run(os.Args[1:])
+	// Create process launch args from the CLI args
+	launchArgs, err := buildLaunchArgs(flag.Args())
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Start the debugger
+	stopDebugger := make(chan bool)
+	debuggerExited := make(chan bool)
+	debugger := service.NewDebugger(stopDebugger)
+	go func() {
+		err := debugger.Run(launchArgs)
+		if err != nil {
+			fmt.Printf("debugger error: %s\n", err)
+		}
+		debuggerExited <- true
+	}()
+
+	// Make a TCP listener
+	listener, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		fmt.Printf("couldn't start listener: %s\n", err)
+		os.Exit(1)
+	}
+	fmt.Printf("started listener on %s\n", listener.Addr())
+
+	// Expose the debugger with a REST service
+	service := rest.NewRESTService(listener, debugger)
+	go service.Run()
+
+	// Start a terminal
+	client := rest.NewRESTClient(listener.Addr().String())
+	term := terminal.New(client)
+	err, status := term.Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Let the server clean up
+	stopDebugger <- true
+	<-debuggerExited
+	fmt.Println("[Hope I was of service hunting your bug!]")
+	glog.Flush()
+	os.Exit(status)
+}
+
+func buildLaunchArgs(args []string) ([]string, error) {
```

Comment:
I actually really don't mind inlining this. It makes the function more verbose but it's all straightline code.